client/dashboard/task-list/tasks/payments/index.js

```diff
@@ -6,30 +6,96 @@
 import { __ } from '@wordpress/i18n';
 import { Fragment, Component } from '@wordpress/element';
 import { compose } from '@wordpress/compose';
-import { filter, noop, keys, pickBy } from 'lodash';
+import { filter, noop, keys, pickBy, difference } from 'lodash';
 import { FormToggle, CheckboxControl } from '@wordpress/components';
 import { Button, TextControl } from 'newspack-components';
+import { withDispatch } from '@wordpress/data';
 
 /**
  * WooCommerce dependencies
  */
 import { getCountryCode } from 'dashboard/utils';
 import { Form, Card, Stepper, List } from '@woocommerce/components';
-import { getHistory, getNewPath } from '@woocommerce/navigation';
+import { getAdminLink, getHistory, getNewPath, getQuery } from '@woocommerce/navigation';
 
 /**
  * Internal dependencies
  */
 import withSelect from 'wc-api/with-select';
 import Plugins from '../steps/plugins';
+import Stripe from './stripe';
+import PayPal from './paypal';
 
 class Payments extends Component {
 	constructor() {
 		super( ...arguments );
+
+		let step = 'choose';
+		let showIndividualConfigs = false;
+		let forceStripeManual = false;
+
+		const query = getQuery();
+		const { options } = this.props;
+		const { woocommerce_onboarding_payments, woocommerce_stripe_settings } = options;
```

Comment:
Thinking on this, I think you are right it will be better to determine from the different settings groups rather than an additional status option.

> #2897 to avoid a bunch of rebasing.

If you are fine with that, I can handle that overhaul there where I think it will be easier, especially after some of the refactoring I did. I'll still wait to merge both of these until they are ready.